http_server/queries/get_info/user.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+function user_select_by_name($pdo, $name) {
+	$stmt = $pdo->prepare('SELECT * FROM users WHERE name = ?');
+	$stmt->bindValue(1, $name, PDO::PARAM_STR);
+	$stmt->execute();
+	$row_count = $stmt->rowCount();
+	if ($row_count < 1) {
+		return false;
+	}
+	// TODO: Add check for more than 1 result and log critical failure if so.
+	$user = $stmt->fetchAll(PDO::FETCH_OBJ);
```

Comment:
Does fetchAll return an array? It looks like fetchObject will work: http://php.net/manual/en/pdostatement.fetchobject.php

Or! Since there doesn't seem to be a better way to get the row count, we could do
```
$stmt->execute();
$results = $stmt->fetchAll();
if (count($results) !== 1) {
  // throw an error
}
$user = $results[0]; // I believe this defaults to being an associative array
return $user;

Reply:
@aaaaaa123456789 would this work?

Reply:
@jacob-grahn the problem with using fetchAll by itself (at least that I've encountered) is that the columns are duplicated (instead of 6 array keys/values, we have 12).  Using fetchObject only gets one result, which _theoretically_ works for this scenario, although I think it would be cleaner to just use `fetchAll(PDO::FETCH_OBJ)` since there isn't really a downside.

To get an associative array, you'd replace `PDO::FETCH_OBJ` with `PDO::FETCH_ASSOC` in `fetchAll()`.

Reply:
@aaaaaa123456789 please correct me if I'm wrong, here.

@jacob-grahn the problem with using fetchAll by itself (at least that I've encountered) is that the columns are duplicated (instead of 6 array keys/values, we have 12).  Using fetchObject only gets one result, which _theoretically_ works for this scenario, although I think it would be cleaner to just use `fetchAll(PDO::FETCH_OBJ)` since there isn't really a downside.

To get an associative array, you'd replace `PDO::FETCH_OBJ` with `PDO::FETCH_ASSOC` in `fetchAll()`.